plasma_framework/contracts/src/exits/payment/PaymentTransactionStateTransitionVerifier.sol

```diff
@@ -0,0 +1,119 @@
+pragma solidity ^0.5.0;
+pragma experimental ABIEncoderV2;
+
+import '../IStateTransitionVerifier.sol';
+import '../payment/PaymentExitDataModel.sol';
+import '../../utils/UtxoPosLib.sol';
+import '../../transactions/WireTransaction.sol';
+import '../../transactions/PaymentTransactionModel.sol';
+import '../../transactions/outputs/PaymentOutputModel.sol';
+
+/*
+* Verifies state transitions for payment transaction
+*/
+contract PaymentTransactionStateTransitionVerifier {
+
+    struct StateTransitionArgs {
+        bytes inFlightTx;
+        bytes[] inputTxs;
+        uint256[] inputUtxosPos;
+    }
+
+    function isCorrectStateTransition(
+        bytes calldata inFlightTx,
+        bytes[] calldata inputTxs,
+        uint256[] calldata inputUtxosPos
+    )
+        external
+        pure
+        returns (bool)
+    {
+        require(inputTxs.length == inputUtxosPos.length, "Input transactions do not match input utxos positions");
+
+        //TODO: refactor that to smaller function as soon as this issue is resolved: https://github.com/ethereum/solidity/issues/6835
+        WireTransaction.Output[] memory inputs = new WireTransaction.Output[](inputTxs.length);
+        for (uint i = 0; i < inputTxs.length; i++) {
+            uint16 outputIndex = UtxoPosLib.outputIndex(UtxoPosLib.UtxoPos(inputUtxosPos[i]));
+            WireTransaction.Output memory output = WireTransaction.getOutput(inputTxs[i], outputIndex);
+            inputs[i] = output;
+        }
+
+        WireTransaction.Output[] memory outputs = new WireTransaction.Output[](inputTxs.length);
+        PaymentTransactionModel.Transaction memory transaction = PaymentTransactionModel.decode(inFlightTx);
+        for (uint i = 0; i < transaction.outputs.length; i++) {
+            outputs[i] = WireTransaction.Output(transaction.outputs[i].amount, transaction.outputs[i].outputGuard, transaction.outputs[i].token);
+        }
+
+        return _isCorrectStateTransition(inputs, outputs);
```

Comment:
Since you're reverting on length mismatch earlier, probably we can consider reverting only strategy to keep consistent or we return false on the earlier. I am on one hand not too much a fan of this reverting strategy as the boolean value become weird, but on the other hand reverting here provides better customized revert message. Also, for unknown reason, revert + return true at the end seems to be the common practice in smart contract (?) 

Anyway, I will leave this to your decision.

Reply:
Done. I went for always returning a boolean. It's more flexible (but on the other hand what can a caller do if it returns false?, will probably always reverse).